File: Cpp/299_Bulls_and_Cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int bulls = 0;
        int count[10] = {0};
        for(int i=0; i<secret.size(); ++i){
            if(secret[i]==guess[i]) ++bulls;
            else {
                count[secret[i]-'0'] += 1;
                count[guess[i]-'0'] -=1;
            }
        }
        
        int wrong = 0;
        for(int i=0; i<10; ++i){
            wrong += max(count[i], 0);
        }
        
        return to_string(bulls) + "A" + to_string(secret.size()-bulls-wrong) + "B";
    }
};

File: Kotlin/299_Bulls_and_Cows.kt
class Solution {
    fun getHint(secret: String, guess: String): String {
        var bulls = 0
        val count = IntArray(10){0}
        for(i in secret.indices) {
            if(guess[i] == secret[i]) {
                ++bulls
            } else {
                count[secret[i]-'0'] += 1
                count[guess[i]-'0'] -=1
            }
        }
        
        var wrong = 0
        count.forEach{
            wrong += maxOf(it, 0)
        }
        
        val stringBuilder = StringBuilder()
        stringBuilder.append(bulls.toString())
        stringBuilder.append("A")
        stringBuilder.append(secret.length - bulls - wrong)
        stringBuilder.append("B")
        
        return stringBuilder.toString()
    }
}